sphinx/cmd/build.py

```diff
@@ -111,9 +112,47 @@ def jobs_argument(value: str) -> int:
             return jobs
 
 
+class ParagraphFormatter(argparse.HelpFormatter):
+    """Wraps help text as a default formatter but keeps paragraps separated."""
+
+    _paragraph_matcher = re.compile(r"\n\n+")
+
+    def _fill_text(self, text: str, width: int, indent: str) -> str:
+        import textwrap
+        result: list[str] = []
+        for p in self._paragraph_matcher.split(text):
+            p = self._whitespace_matcher.sub(' ', p).strip()
+            p = textwrap.fill(p, width,
+                              initial_indent=indent,
+                              subsequent_indent=indent)
+            result.append(p)
+        return '\n\n'.join(result)
+
+
+class ArgParser(argparse.ArgumentParser):
+    """Wraps standard ArgumentParser to add sphinx-specefic flags to help."""
+
+    def print_help(self, file: IO[str] | None = None) -> None:
+        from gettext import gettext as _
+        # we inject -M flag action to positionals before printing help
+        # so that there is no risk of side effects on actual execution
+        for action_group in self._action_groups:
+            if action_group.title != _('positional arguments'):
+                continue
+            m_flag = argparse.Action(
+                ["-M"], "BUILDER",  # ugly but works
+                help=__('please refer to usage and main help section')
+            )
+            action_group._group_actions.insert(0, m_flag)
```

Comment:
I would like to push back on this specific comment. `argparse` is what is used by the original code and there is no way to replace it without changing a whole lot of things. Since @AA-Turner seems to build a new version of a cli (at least it looks like that from his last commits), there isn't much sense to me to reinvent a second bicycle.
As to reverting back the attributes after the injection: it is performed specifically in a call of `print_help`. This call always ends with a termination of a program so there is no point in reverting the changes. It is exactly the reason I have implemented it in the call itself.
As for your other comments - thank you for your input, I will try to implement and push those as soon as I can.

Reply:
Mmh. Let's live with this hack then.

We prefer not to have force pushes since it's hard to review commits. We always squash merge at the end so you just push commits one by one. I'd suggest rebasing locally if you want to squash your own commits but avoid force-pushing in general.